obspy/io/resp/nrl.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+'''
+nrl.py for navigating, and picking responses from the
+Nominal Response Library.
+
+:copyright:
+    Lloyd Carothers IRIS/PASSCAL, 2016
+:license:
+    GNU Lesser General Public License, Version 3
+    (http://www.gnu.org/copyleft/lesser.html)
+'''
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+# Urllib for py 2 and 3
+from future.standard_library import install_aliases
+install_aliases()  # NOQA
+from urllib.parse import urljoin
+from urllib.request import urlopen
+from os.path import join, dirname
+from ConfigParser import SafeConfigParser
+
+
+class NRL:
+    """
+    Object representing the Nominal Response library.
+    Can be Accessed online from the DMC or with a local copy
+    Decision tree: Usually!!!! depth of tree and order can change!
+    Sensor, make, model, period, sensitivity
+    Datalogger, make, model, gain, sample rate
+
+    The NRL has txt desision files formatted as windows ini
+
+    """
+    URLroot = 'http://ds.iris.edu/NRL/'
+    index = 'index.txt'
+    # Placholder for sample rate and gain
+    SR = object()
+    GAIN = object()
+    dl_shortcuts = {
+        'rt130': ['REF TEK', 'RT 130 & 130-SMA', GAIN, SR],
+        'q330': ['Quanterra', 'Q330SR', GAIN, SR, 'LINEAR AT ALL SPS']}
+    sensor_shortcuts = {
+        'cmg3t': ['Guralp', 'CMG-3T', '120s - 50 Hz', '1500'],
+        'trillium_240_1': [
+            'Nanometrics', 'Trillium 240', '1 - serial numbers < 400'],
+        'l22': ['Sercel/Mark Products', 'L-22D', '5470 Ohms', '20000 Ohms'],
+        'sts2_g3': [
+            'Streckeisen', 'STS-2', '1500', '3 - installed 04/97 to present']
+        }
+
+    def __init__(self, local=True, root=None):
```

Comment:
Here's actually a code snippet that would work like I had this in mind (sorry, didn't have much time when writing that comment above):
```python
class NRL(object):
    def __new__(cls, root='http://ds.iris.edu/NRL'):
        if cls is not NRL:
            return super(NRL, cls).__new__(cls)
        if "://" in root:
            return RemoteNRL(root=root)
        else:
            return LocalNRL(root=root)

    def __init__(self, root='http://ds.iris.edu/NRL'):
        self.root = root

    # common methods here


class LocalNRL(NRL):
    def _check(self):
        import os
        if not os.path.exists(self.root):
            raise ValueError()


class RemoteNRL(NRL):
    def _check(self):
        # e.g. check if URL can be loaded here
        pass
```

```python
from nrl_example import NRL, LocalNRL, RemoteNRL

nrl = NRL()
assert isinstance(nrl, NRL)
assert isinstance(nrl, RemoteNRL)
assert not isinstance(nrl, LocalNRL)
print(type(nrl))

nrl = NRL(root='/tmp')
assert isinstance(nrl, NRL)
assert isinstance(nrl, LocalNRL)
assert not isinstance(nrl, RemoteNRL)
print(type(nrl))
```
```
<class 'nrl_example.RemoteNRL'>
<class 'nrl_example.LocalNRL'>
```

I think this might be clearer structured than the current..
```python
        if local:
            # use local copy of NRL on filesystem
            self.read_ini = self.read_fs_ini
            self.read_resp = self.read_resp_local
            self.join = self.join_fs
            if root is None:
                raise Exception('Root path of NRL required')
            else:
                self.root = root
        else:
            # use online, urls etc
            self.read_ini = self.read_url_ini
            self.read_resp = self.read_resp_url
            self.join = urljoin
            if root is None:
                self.root = self.URLroot
            else:
```